ste/sender-blockBlobFromLocal.go

```diff
@@ -86,26 +83,18 @@ func (u *blockBlobUploader) generatePutWholeBlob(id common.ChunkID, blockIndex i
 
 	return createSendToRemoteChunkFunc(u.jptm, id, func() {
 		jptm := u.jptm
+		encodedBlockID := u.generateEncodedBlockID()
+		u.setBlockID(blockIndex, encodedBlockID)
 
 		// Upload the blob
 		jptm.LogChunkStatus(id, common.EWaitReason.Body())
 		var err error
 		if jptm.Info().SourceSize == 0 {
-			_, err = u.destBlockBlobURL.Upload(jptm.Context(), bytes.NewReader(nil), u.headersToApply, u.metadataToApply, azblob.BlobAccessConditions{})
+			u.blockIDs = make([]string, 0) // Empty block ID list
 		} else {
-			// File with content
-
-			// Get the MD5 that was computed as we read the file
-			md5Hash, ok := <-u.md5Channel
-			if !ok {
-				jptm.FailActiveUpload("Getting hash", errNoHash)
-				return
-			}
-			u.headersToApply.ContentMD5 = md5Hash
```

Comment:
Why removing?

Reply:
In the old changes, this would've caused two pulls from the MD5 channel.